universal-login-sdk/lib/sdk.ts

```diff
@@ -60,78 +60,21 @@ class UniversalLoginSDK {
     return this.futureWalletFactory!.createFutureWallet();
   }
 
-  async addKey(
-    to: string,
-    publicKey: string,
-    privateKey: string,
-    transactionDetails: Message,
-    keyPurpose = MANAGEMENT_KEY,
-  ) {
-    const key = publicKey;
-    const data = new utils.Interface(WalletContract.interface).functions.addKey.encode([key, keyPurpose]);
-    const message = {
-      ...transactionDetails,
-      to,
-      from: to,
-      data,
-    };
-    return this.execute(message, privateKey);
+  async addKey(to: string, publicKey: string, privateKey: string, transactionDetails: Message, keyPurpose = MANAGEMENT_KEY) {
+    return this.selfExecute(to, 'addKey', [publicKey, keyPurpose], privateKey, transactionDetails);
```

Comment:
Just to be consistent with addKeys(): keyPurpose -> keyRole?

Reply:
`return this.selfExecute(to, 'addKeys', [[publicKey], [keyRole]], privateKey, transactionDetails);`